### PR TITLE
Facebook Login

### LINKED ISSUE #36: Sign In Page
As a user I should be able to sign in so that my personal pins are not accessible to other people.

### PR DESCRIPTION
**Pull request by tiffeli**
The last pull request was probably not working for you guys because of the react native upgrade/changes to config files. I just redid FB login from an upgraded branch.

you must npm install since it's using a new package 'react-native-facebook-login'

The login feature:  specifies user!
1) adds user to firebase in the form we discussed
"" : {"id"...., "picture":... etc}
-you can view the full structure in the database
2) calls action which updates he state so state.user = {} changes to the same obj data as firebase!
3) pins to users own database and goes to users specific pins
4) if logged in already, will go to view directly
Resolves #36

### CODE REVIEW

File: app/lib/db/db.js [hunk 1]
```diff
@@ -15,4 +16,4 @@ export const changeUser = function( uid ) {
     myCurrLoc = currLoc.child( uid );
     userData = user.child( 'pins' );
     userRecent = user.child( 'recent' );
-}
\ No newline at end of file
+}
```
Comment (jake-shasteen) [inline app/lib/db/db.js]: Nice catch!

File: app/containers/container_FBLogin.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  render() {
+  const { user, action } = this.props;
+    var _this = this;
+    return (
+      <View style={styles.loginContainer}>
+
+        <FBLogin style={{ marginBottom: 10, }}
+          permissions={["email","user_friends"]}
+          onLogin={function(data){
+            console.log("*******Logged in! and this is props, this is data cred", data.credentials);
```
Comment (jake-shasteen) [inline app/containers/container_FBLogin.js]: We can toss this debugging console.log

File: app/actions/action_user.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+};
+
+export const firebase_check = (userCredentials) => {
+  let id = userCredentials.userId;
+  let token = userCredentials.token;
+  let api = "https://graph.facebook.com/v2.3/"+id+"?fields=name,email,picture&access_token="+token;
+  function checkIfUserExists(userId, callback) {
+    ref.once('value', function(snapshot) {
+    let userExistsBool = snapshot.hasChild(userId);
+    console.log(userExistsBool, 'checkIfUserExists bool')
```
Comment (jake-shasteen) [inline app/actions/action_user.js]: We can get rid of this testing console.log

File: app/containers/container_FBLogin.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+          onCancel={function(){
+            console.log("******User cancelled.");
+          }}
+          onPermissionsMissing={function(data){
+            console.log("*******Check permissions!");
+            console.log(data);
+          }}
+        />
+
+        <Text>{ user ? user.token : "N/A" }</Text>
```
Comment (jake-shasteen) [inline app/containers/container_FBLogin.js]: I can't find where this actually appears in testing the app. What is this for?

File: app/containers/container_FBLogin.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  bottomBump: {
+    marginBottom: 15,
+  },
+});
+
+function mapStateToProps(state) {
+  return {
+    user: state.user,
+    photo: state.user.photo,
+    info: state.user.info
```
Comment (jake-shasteen) [inline app/containers/container_FBLogin.js]: These don't seem to be used anywhere

File: app/containers/container_FBLogin.js (hunk 4 above)
Comment (tiffeli) [inline app/containers/container_FBLogin.js]: @jake-shasteen this was for my own testing before I changed state--> redux state. I'll take this out!

**Pull request merged**